### PR TITLE
Db schema update, SetAccountLabel RPC impl, test refactor

### LINKED ISSUE #15: Race conditions when testing
Sometimes seems like tests generate race conditions. Further info will follow.

### LINKED ISSUE #44: Account namespace and label
Following the discussion of #42, we should update the protos by renaming the `Name` field of the `Create*Account` request message to `Label` and make it optional in the service instead of being required.
Also, we should update the response message by returning the `AccountInfo` of the newly created account.

Internally, we should rename the related domain's actual `Name` field to `Label` and add a new one `Namespace`.

The namespace is something immutable and assigned by the service to every account. It could be a string like `<root_path_purpose>-account-<derivation index>`, for example `84-account-0` for account 0 of wallet with root path `m/84'/<network>'`. The namespace must be used by the repo to identify an account uniquely.

The label, instead, is something mutable and set by the user optionally. If a label is set, the user/client can refer to the account with it, otherwise the namespace must be used.

We should add also a new rpc `SetAccountLabel` to allow one to change account label at will.

### PR DESCRIPTION
**Pull request by sekulicd**
Changes:
1. Main thing was to add primary key namespace column in account table and label column (account table as well) 
2. Tables account_script_info, tx_input_account,utxo are updated with ref(fk_account_namespace) to account table
3. Few commands were added to the Makefile that are used to work easier with db, vet_db -> checks if up and down mig scripts are ok
4. DB layer tests are moved to the new location /test and test suites are added as necessity to isolate state for each test case (run before/after code where needed) and to prepare/clean db for each test suite, also whit this some issues with race conditions disappeared 
5. RPC SetAccountLabel is added that updates account label
6. Other stuff that are changed in other layers(app, interface, cli) are just renaming name to namespace and adding label
7. Race issue is fixed with [188f55e](https://github.com/vulpemventures/ocean/pull/45/commits/188f55e6e76b4107dcbb7a5213c313375691a3c1), isLocked was not protected with mutex in sendReport method.

This closes #44 
This closes #15

@tiero @altafan please review.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -99,10 +121,39 @@ pgcreatetestdb:
...
+
+## mig_down_yes: apply down migration without prompt
+mig_down_yes:
+	@echo "migration down..."
+	@"yes" | migrate -database "postgres://root:secret@localhost:5432/oceand-db?sslmode=disable" -path ./internal/infrastructure/storage/db/postgres/migration/ down
+
+## vet_db: check if mig_up and mig_down are ok
+vet_db: recreatedb mig_up mig_down_yes
+	@echo "vet db migration scripts..."
+
```
Comment (altafan) [inline Makefile:156]: what are these commands used for?

File: api-spec/protobuf/ocean/v1/account.proto [hunk 2]
```diff
@@ -97,8 +84,8 @@ message SetAccountTemplateRequest {
 message SetAccountTemplateResponse {}
 
 message DeriveAddressesRequest{
-  // Account name.
-  string account_name = 1;
+  // Account namespace.
+  string namespace = 1;
```
Comment (altafan) [inline api-spec/protobuf/ocean/v1/account.proto:88]: I would call this field `name` to allow user referring to an account with namespace or label.
```suggestion
  // Account namespace or label.
  string name = 1;
```

Comment (altafan) [reply api-spec/protobuf/ocean/v1/account.proto:88]: This change should be made to all RPCs

File: api-spec/protobuf/ocean/v1/types.proto [hunk 3]
```diff
@@ -12,14 +12,16 @@ message BuildInfo {
 }
 
 message AccountInfo {
-  // Account name.
-  string name = 1;
+  // Label name.
+  string label = 1;
```
Comment (altafan) [inline api-spec/protobuf/ocean/v1/types.proto:16]: nit: we could switch the fields to respect the order of importance

File: cmd/ocean/account.go [hunk 4]
```diff
@@ -56,6 +57,12 @@ var (
 			"The wallet will loose track of every derived address and utxo history",
 		RunE: accountDelete,
 	}
+	accountSetLabelCmd = &cobra.Command{
+		Use:   "label",
+		Short: "set accounts label",
```
Comment (altafan) [inline cmd/ocean/account.go:62]:
```suggestion
		Short: "set account label",
```

File: cmd/ocean/account.go [hunk 5]
```diff
@@ -10,7 +10,8 @@ import (
 )
 
 var (
-	accountName     string
+	label           string
+	namespace       string
```
Comment (altafan) [inline cmd/ocean/account.go:14]: Let's revert these changes. An account name can be either a namespace or label.

```suggestion
	accountName           string
```

File: cmd/ocean/account.go [hunk 6]
```diff
@@ -56,6 +57,12 @@ var (
 			"The wallet will loose track of every derived address and utxo history",
 		RunE: accountDelete,
 	}
+	accountSetLabelCmd = &cobra.Command{
```
Comment (altafan) [inline cmd/ocean/account.go:60]: From the user POV, this command should look like:
```
ocean account label <label> --account-name <namespace>
```

This way we don't need an extra flag to retrieve the value of the label.

File: cmd/ocean/account.go [hunk 7]
```diff
@@ -245,3 +252,22 @@ func accountDelete(cmd *cobra.Command, _ []string) error {
 	fmt.Println("account deleted")
 	return nil
 }
+
+func accountSetLabel(cmd *cobra.Command, _ []string) error {
```
Comment (altafan) [inline cmd/ocean/account.go:256]: according to the changes above, here the label should be retrieved from args (which are now ignored) instead of flags.

File: internal/core/application/account_service.go [hunk 8]
```diff
@@ -10,6 +10,10 @@ import (
 	"github.com/vulpemventures/ocean/internal/core/ports"
 )
 
+const (
+	bip84RootPathPurpose = "84"
+)
```
Comment (altafan) [inline internal/core/application/account_service.go:15]: Not needed. Remove it, please.

File: internal/core/application/account_service.go [hunk 9]
```diff
@@ -54,14 +58,14 @@ func NewAccountService(
...
-	ctx context.Context, accountName string,
+	ctx context.Context, label string,
 ) (*AccountInfo, error) {
 	_, birthdayBlockHeight, err := as.bcScanner.GetLatestBlock()
 	if err != nil {
 		return nil, err
 	}
 	accountInfo, err := as.repoManager.WalletRepository().CreateAccount(
-		ctx, accountName, birthdayBlockHeight,
+		ctx, bip84RootPathPurpose, label, birthdayBlockHeight,
```
Comment (altafan) [inline internal/core/application/account_service.go:68]: The following is an example of an account's namespace for account 0 of a BIP84 wallet:  `bip84-account0`.

Generally, the root path's purpose should be extracted from the [wallet's derivation root path](https://github.com/vulpemventures/ocean/blob/master/internal/core/domain/wallet.go#L55).
Also, the namespace should be generated by the domain and not provided as arg like the label.

File: internal/core/application/account_service.go [hunk 10]
```diff
@@ -70,90 +74,90 @@ func (as *AccountService) CreateAccountBIP44(
 }
 
 func (as *AccountService) DeriveAddressesForAccount(
-	ctx context.Context, accountName string, numOfAddresses uint64,
+	ctx context.Context, namespace string, numOfAddresses uint64,
```
Comment (altafan) [inline internal/core/application/account_service.go:77]: According to the changes above, this should become a generic `name`.
```suggestion
	ctx context.Context, name string, numOfAddresses uint64,
```

File: internal/core/application/account_service.go [hunk 11]
```diff
@@ -70,90 +74,90 @@ func (as *AccountService) CreateAccountBIP44(
...
 ) (AddressesInfo, error) {
 	if numOfAddresses == 0 {
 		numOfAddresses = 1
 	}
 
 	addressesInfo, err := as.repoManager.WalletRepository().
-		DeriveNextExternalAddressesForAccount(ctx, accountName, numOfAddresses)
+		DeriveNextExternalAddressesForAccount(ctx, namespace, numOfAddresses)
```
Comment (altafan) [inline internal/core/application/account_service.go:84]: Finally, also the repo should allow referring to an account via label or namespace and this should be completely abstracted for the app layer so that we just need to change a variable's name:
```suggestion
		DeriveNextExternalAddressesForAccount(ctx, name, numOfAddresses)
```

File: internal/core/domain/utxo.go [hunk 12]
```diff
@@ -32,14 +32,14 @@ func (k UtxoKey) String() string {
...
-	AssetBlinder    []byte
-	AccountName     string
-	SpentStatus     UtxoStatus
-	ConfirmedStatus UtxoStatus
+	Value              uint64
+	Asset              string
+	Script             []byte
+	ValueBlinder       []byte
+	AssetBlinder       []byte
+	FkAccountNamespace string
```
Comment (altafan) [inline internal/core/domain/utxo.go:40]: Foreign keys are a concept of relational DBs only, and since domains are totally unrelated to repo implementations, it doesn't really make sense to use this kind of name IMO.

We can either keep this called `AccountName` to prevent making other renaming changes, or we can use `AccountNamespace`, but please, remove the `Fk` prefix.

File: internal/core/domain/utxo_repository.go [hunk 13]
```diff
@@ -74,7 +74,7 @@ type UtxoRepository interface {
 	UnlockUtxos(ctx context.Context, utxoKeys []UtxoKey) (int, error)
 	// DeleteUtxosForAccount deletes every utxo associated to the given account
 	// from the repository.
-	DeleteUtxosForAccount(ctx context.Context, accountName string) error
+	DeleteUtxosForAccount(ctx context.Context, namespace string) error
```
Comment (altafan) [inline internal/core/domain/utxo_repository.go:77]: The changes to this file are not so relevant. Can we revert them except for this one? Let's align it with the existing ones:
```suggestion
	DeleteUtxosForAccount(ctx context.Context, account string) error
```

File: internal/core/domain/wallet.go [hunk 14]
```diff
@@ -16,6 +16,9 @@ import (
 const (
 	externalChain = 0
 	internalChain = 1
+
+	// namespaceFormat <root_path_purpose>-account-<derivation index>
+	namespaceFormat = "%v-account-%v"
```
Comment (altafan) [inline internal/core/domain/wallet.go:21]:
```suggestion
	namespaceFormat = "bip%v-account%v"
```

File: internal/core/domain/wallet.go [hunk 15]
```diff
@@ -49,15 +52,15 @@ type AddressInfo struct {
...
-	AccountKeysByName   map[string]string
-	NextAccountIndex    uint32
+	EncryptedMnemonic      []byte
+	PasswordHash           []byte
+	BirthdayBlockHeight    uint32
+	RootPath               string
+	NetworkName            string
+	AccountsByKey          map[string]*Account
+	AccountKeysByIndex     map[uint32]string
+	AccountKeysByNamespace map[string]string
```
Comment (altafan) [inline internal/core/domain/wallet.go:62]: All these mappings were initially thought for referring to accounts via derivation index or name, and the account key was like a hash of index+name.

I think we can simplify this by having 2 maps `AccountsByNamesspace map[string]*Account` and `AccountNamespacesByLabel map[string]string`.

File: internal/core/domain/wallet_account.go [hunk 16]
```diff
@@ -10,18 +10,19 @@ import (
 
 // AccountKey holds the unique info of an account: name and HD index.
 type AccountKey struct {
-	Name  string
-	Index uint32
+	Namespace string
+	Index     uint32
 }
```
Comment (altafan) [inline internal/core/domain/wallet_account.go:15]: This struct was meant originally to let the user referring to an account via index or name, and it was helpful to generate the account key ad a hash of index+name.
It can be removed now since we are going to refer to accounts via namespace or label.

File: internal/core/domain/wallet_account.go [hunk 17]
```diff
@@ -10,18 +10,19 @@ import (
...
 func (ak *AccountKey) String() string {
-	key := btcutil.Hash160([]byte(fmt.Sprintf("%s%d", ak.Name, ak.Index)))
+	key := btcutil.Hash160([]byte(fmt.Sprintf("%s%d", ak.Namespace, ak.Index)))
 	return hex.EncodeToString(key[:6])
 }
 
 // AccountInfo holds basic info about an account.
 type AccountInfo struct {
 	Key            AccountKey
+	Label          string
```
Comment (altafan) [inline internal/core/domain/wallet_account.go:25]:
```suggestion
        Index uint32
	Namespace string
	Label          string
```

File: internal/core/domain/wallet_repository.go [hunk 18]
```diff
@@ -51,8 +51,8 @@ type WalletRepository interface {
 	// UnlockWallet attempts to update the status of the Wallet to "unlocked".
 	// Generates a WalletUnlocked event if successfull.
 	UnlockWallet(ctx context.Context, password string) error
-	// LockkWallet updates the status of the Wallet to "locked".
-	// Generates a WalletLocked event if successfull.
+	// LockWallet updates the status of the Wallet to "locked".
+	//Generates a WalletLocked event if successfull.
```
Comment (altafan) [inline internal/core/domain/wallet_repository.go:55]: Revert changes to this line:
```suggestion
	// Generates a WalletLocked event if successfull.
```

File: internal/core/domain/wallet_repository.go [hunk 19]
```diff
@@ -63,23 +63,23 @@ type WalletRepository interface {
 	// its basic info.
 	// Generates a WalletAccountCreated event if successfull.
 	CreateAccount(
-		ctx context.Context, accountName string, birthdayBlock uint32,
+		ctx context.Context, rootPathPurpose, label string, birthdayBlock uint32,
```
Comment (altafan) [inline internal/core/domain/wallet_repository.go:66]: Not necessary to provide `rootPathPurpose` as arg since it can be extracted from wallet's RootPath.
```suggestion
		ctx context.Context, label string, birthdayBlock uint32,
```

File: internal/core/domain/wallet_repository.go [hunk 20]
```diff
@@ -63,23 +63,23 @@ type WalletRepository interface {
...
 	) (*AccountInfo, error)
 	// DeriveNextExternalAddressesForAccount returns one or more new receiving
 	// addresses for the given account.
 	// Generates a WalletAccountAddressesDerived event if successfull.
 	DeriveNextExternalAddressesForAccount(
-		ctx context.Context, accountName string, numOfAddresses uint64,
+		ctx context.Context, namespace string, numOfAddresses uint64,
```
Comment (altafan) [inline internal/core/domain/wallet_repository.go:72]: either revert the changes to this and the other APIs, or change to generic `account`.

File: internal/core/domain/wallet_account.go (hunk 17 above)
Comment (altafan) [reply internal/core/domain/wallet_account.go:25]: Also, a field `Index uint32` should be added to `Account`.

File: internal/infrastructure/blockchain-scanner/electrum/db.go [hunk 21]
```diff
@@ -23,10 +23,10 @@ func (t dbEventType) String() string {
 }
 
 type dbEvent struct {
-	eventType  dbEventType
-	tx         txInfo
-	account    string
-	scriptHash string
+	eventType        dbEventType
```
Comment (altafan) [inline internal/infrastructure/blockchain-scanner/electrum/db.go:26]: The changes to this file are not so relevant, please revert them.

File: internal/infrastructure/blockchain-scanner/electrum/service.go [hunk 22]
```diff
@@ -110,37 +110,37 @@ func (s *service) Stop() {
 }
 
 func (s *service) WatchForAccount(
-	accountName string, _ uint32,
+	namespace string, _ uint32,
```
Comment (altafan) [inline internal/infrastructure/blockchain-scanner/electrum/service.go:113]: same here.

File: internal/infrastructure/blockchain-scanner/electrum/tcp_client.go [hunk 23]
```diff
@@ -159,61 +159,61 @@ func (c *tcpClient) subscribeForBlocks() {
 }
 
 func (c *tcpClient) subscribeForAccount(
-	accountName string, addresses []domain.AddressInfo,
+	accountNamespace string, addresses []domain.AddressInfo,
```
Comment (altafan) [inline internal/infrastructure/blockchain-scanner/electrum/tcp_client.go:162]: same here.

File: internal/infrastructure/blockchain-scanner/electrum/ws_client.go [hunk 24]
```diff
@@ -160,56 +160,56 @@ func (c *wsClient) subscribeForBlocks() {
 }
 
 func (c *wsClient) subscribeForAccount(
-	accountName string, addresses []domain.AddressInfo,
+	accountNamespace string, addresses []domain.AddressInfo,
```
Comment (altafan) [inline internal/infrastructure/blockchain-scanner/electrum/ws_client.go:163]: same here.

File: internal/infrastructure/blockchain-scanner/elements/scanner.go [hunk 25]
```diff
@@ -31,7 +31,7 @@ type scannerService struct {
 }
 
 func newScannerSvc(
-	accountName string,
+	accountNamespace string,
```
Comment (altafan) [inline internal/infrastructure/blockchain-scanner/elements/scanner.go:34]: same here.

File: internal/infrastructure/blockchain-scanner/elements/service.go [hunk 26]
```diff
@@ -90,34 +90,34 @@ func (s *service) Start() {}
 
 func (s *service) Stop() {}
 
-func (s *service) GetUtxoChannel(accountName string) chan []*domain.Utxo {
-	scannerSvc := s.getOrCreateScanner(accountName, 0)
+func (s *service) GetUtxoChannel(namespace string) chan []*domain.Utxo {
```
Comment (altafan) [inline internal/infrastructure/blockchain-scanner/elements/service.go:93]: same here.

File: internal/infrastructure/blockchain-scanner/neutrino/scanner.go [hunk 27]
```diff
@@ -18,7 +18,7 @@ import (
 )
 
 type scannerService struct {
-	accountName         string
+	accountNamespace    string
```
Comment (altafan) [inline internal/infrastructure/blockchain-scanner/neutrino/scanner.go:21]: same here.

File: internal/infrastructure/storage/db/postgres/migration/20230112084523_accoun_table_update.down.sql [hunk 28]
```diff
@@ -0,0 +1,19 @@
...
+
+ALTER TABLE account DROP COLUMN namespace;
+
+ALTER TABLE account DROP COLUMN label;
+
+ALTER TABLE account ADD COLUMN name VARCHAR(50) PRIMARY KEY;
+
+ALTER TABLE account_script_info ADD COLUMN fk_account_name VARCHAR(50);
+
+ALTER TABLE account_script_info ADD FOREIGN KEY (fk_account_name) REFERENCES account(name);
```
Comment (altafan) [inline internal/infrastructure/storage/db/postgres/migration/20230112084523_accoun_table_update.down.sql:15]: Please, add `ON DELETE CASCADE`  on foreign key fields so that when a record in deleted, all its references in this and other tables are deteled as well.

File: internal/infrastructure/storage/db/postgres/sqlc/query.sql [hunk 29]
```diff
@@ -4,36 +4,39 @@ INSERT INTO wallet(id, encrypted_mnemonic,password_hash,birthday_block_height,ro
...
-LEFT JOIN account_script_info asi on a.name = asi.fk_account_name
+LEFT JOIN account_script_info asi on a.namespace = asi.fk_account_namespace
 WHERE w.id = $1;
 
 -- name: UpdateWallet :one
 UPDATE wallet SET encrypted_mnemonic = $2, password_hash = $3, birthday_block_height = $4, root_path = $5, network_name = $6, next_account_index = $7 WHERE id = $1 RETURNING *;
 
 -- name: GetAccount :one
-SELECT * FROM account WHERE name = $1;
+SELECT * FROM account WHERE namespace = $1;
```
Comment (altafan) [inline internal/infrastructure/storage/db/postgres/sqlc/query.sql:19]: Should be like this to support namespace/label, shouldn't it?
```suggestion
SELECT * FROM account WHERE namespace = $1 OR label = $1;
```

Comment (altafan) [reply internal/infrastructure/storage/db/postgres/sqlc/query.sql:19]: This change should be made to all queries referring account only via namespace so far

File: internal/infrastructure/storage/db/postgres/wallet_repository.go [hunk 30]
```diff
@@ -169,7 +169,7 @@ func (w *walletRepositoryPg) UpdateWallet(
 	// insert account scripts as well
 	for _, account := range updatedWallet.AccountsByKey {
 		newAccount := false
-		_, err := querierWithTx.GetAccount(ctx, account.Info.Key.Name)
+		_, err := querierWithTx.GetAccount(ctx, account.Info.Key.Namespace)
 		if err != nil {
 			if err.Error() == pgxNoRows {
 				newAccount = true
```
Comment (altafan) [inline internal/infrastructure/storage/db/postgres/wallet_repository.go:175]: Sorry, I didn't notice this before but this is not the correct behavior, let's fix it now: in an Update* API, if the entry is not found we don't want to add it but rather we must return a not found error. We have Create/Add APIs for that purpose.

File: internal/interfaces/grpc/handler/account.go [hunk 31]
```diff
@@ -21,20 +21,19 @@ func NewAccountHandler(appSvc *application.AccountService) pb.AccountServiceServ
...
 		return nil, err
 	}
+
 	return &pb.CreateAccountBIP44Response{
-		AccountName:    accountInfo.Key.Name,
-		AccountIndex:   accountInfo.Key.Index,
-		Xpub:           accountInfo.Xpub,
-		DerivationPath: accountInfo.DerivationPath,
+		AccountInfo: &pb.AccountInfo{
+			Label:          req.GetLabel(),
```
Comment (altafan) [inline internal/interfaces/grpc/handler/account.go:31]: let's source the label from account's info:
```suggestion
			Label:          accountInfo.Label,
```

File: internal/interfaces/grpc/handler/account.go [hunk 32]
```diff
@@ -59,14 +58,14 @@ func (a *account) SetAccountTemplate(
 func (a *account) DeriveAddresses(
 	ctx context.Context, req *pb.DeriveAddressesRequest,
 ) (*pb.DeriveAddressesResponse, error) {
-	name, err := parseAccountName(req.GetAccountName())
+	namespace, err := parseAccountNamespace(req.GetNamespace())
```
Comment (altafan) [inline internal/interfaces/grpc/handler/account.go:61]: The changes to this and all following APIs could be basically all reverted since they are not so relevant.

File: api-spec/protobuf/ocean/v1/types.proto [hunk 33]
```diff
@@ -12,14 +12,16 @@ message BuildInfo {
...
   // Account index.
   uint32 index = 2;
```
Comment (altafan) [inline api-spec/protobuf/ocean/v1/types.proto:18]: Let's remove the index from the account's info. It's already encoded in the namespace and derivation path.

File: api-spec/protobuf/ocean/v1/types.proto [hunk 34]
```diff
@@ -73,8 +75,8 @@ message Output {
 }
 
 message Utxos {
-  // Account name.
-  string account_name = 1;
+  // Account namespace.
+  string namespace = 1;
```
Comment (altafan) [inline api-spec/protobuf/ocean/v1/types.proto:79]: Either we revert this or we change it to a more generic `account`. The idea is not to remove that prefix because otherwise, it wouldn't be so clear what this namespace was without reading a doc.

File: api-spec/protobuf/ocean/v1/types.proto [hunk 35]
```diff
@@ -100,8 +102,8 @@ message Utxo {
   string asset_blinder = 6;
   // Value blinder for confidential utxo.
   string value_blinder = 7;
-  // Name of the account owning the utxo.
-  string account_name = 8;
+  // Namespace of the account owning the utxo.
+  string namespace = 8;
```
Comment (altafan) [inline api-spec/protobuf/ocean/v1/types.proto:106]: same for this field.

File: internal/interfaces/grpc/handler/account.go [hunk 36]
```diff
@@ -141,39 +140,55 @@ func (a *account) Balance(
...
 }
+
+func (a *account) SetAccountLabel(
+	ctx context.Context,
+	req *pb.SetAccountLabelRequest,
+) (*pb.SetAccountLabelResponse, error) {
+	namespace, err := parseAccountNamespace(req.GetNamespace())
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
```
Comment (altafan) [inline internal/interfaces/grpc/handler/account.go:187]: According to prev required changes, this should be (note that we are currently missing to validate the label):
```suggestion
	namespace, err := parseAccountName(req.GetNamespace())
	if err != nil {
		return nil, status.Error(codes.InvalidArgument, err.Error())
	}
	label, err := parseAccountName(req.GetLabel())
	if err != nil {
		return nil, status.Error(codes.InvalidArgument, err.Error())
	}
```

Comment (altafan) [reply internal/interfaces/grpc/handler/account.go:187]: since `parseAccountName()` makes sure that the given string is not empty, we can re-use it for both namespace and label. This is the only check we have to make at interface layer to validate request arguments. The app layer will tell if they are valid or not after using them eventually.

File: internal/interfaces/grpc/handler/account.go [hunk 37]
```diff
@@ -141,39 +140,55 @@ func (a *account) Balance(
...
+
+	if err := a.appSvc.SetAccountLabel(ctx, namespace, req.GetLabel()); err != nil {
```
Comment (altafan) [inline internal/interfaces/grpc/handler/account.go:189]: According to prev required changes, this should look like (note that we are missing to validate the given label): 

```suggestion
	namespace, err := parseAccountName(req.GetNamespace())
	if err != nil {
		return nil, status.Error(codes.InvalidArgument, fmt.Errorf("missing namespace"))
	}
	label, err := parseAccountName(req.GetLabel())
	if err != nil {
		return nil, status.Error(codes.InvalidArgument, fmt.Errorf("missing label"))
	}
	if err := a.appSvc.SetAccountLabel(ctx, namespace, label); err != nil {
```

since parseAccountName() makes sure that the given string is not empty, we can re-use it for both namespace and label. This is the only check we have to make at interface layer to validate request arguments. The app layer will tell if they are valid or not after using them eventually.

File: api-spec/protobuf/ocean/v1/notification.proto [hunk 38]
```diff
@@ -37,8 +37,8 @@ message TransactionNotificationsRequest{}
 message TransactionNotificationsResponse{
   // Tx event type.
   TxEventType event_type = 1;
-  // Account names.
-  repeated string account_names = 2;
+  // Accounts namespaces.
+  repeated string namespaces = 2;
```
Comment (altafan) [inline api-spec/protobuf/ocean/v1/notification.proto:41]: Please revert these changes. From a user POV, a namespace is an account name just like a label.

Comment (altafan) [reply api-spec/protobuf/ocean/v1/notification.proto:41]: Maybe better to call this field just `accounts`?

**Comment by sekulicd** [comment]
I see most of comments are related to the request to abstract away from user namespace/label columns into one called(account, accountName, name). For UX perspective this is better option but it adds bit of complexity to the backend, since we dont know wether user passed namespace or label we need to have two maps accountsPerNamespace and accountsPerLaber in order to support search "1 in 2(one field that could mean two things)", we need to update repositories as well.
Field previously called accountName was all over the code base and wanted to be consistent, so the code is more readable and that person who reads it can recognise same arg name everywhere wether it is in scanners or repos or somewhere else.

So se this was already big change, it we want to go with the requirement to allow user to search in one filed using namespace/label please suggest how we want to call and i suggest to use it everywhere.

WDYT @tiero @altafan ?

**Comment by tiero** [comment]
Would starting with namespace only enough for time being? Will refactor later to add label?

**Comment by altafan** [comment]
> So se this was already big change, it we want to go with the requirement to allow user to search in one filed using namespace/label please suggest how we want to call and i suggest to use it everywhere.

Let's refer internally to an account just as `account`, like for example in utxos, transactions, scanners etc.
Let's use instead `name` to allow the user refer to an account (protobuf), whether he's passing a namespace or label. Let's keep using the flag --account-name without changing it.
The only place where we need to make an explicit distinction between namespace and label is in the creation of the wallet and for setting an account label.

File: Makefile (hunk 1 above)
Comment (sekulicd) [reply Makefile:156]: mig_up -> create tables based on migration scripts 
mig_down -> executes down migration scripts, basically drop tables
vet_db -> checks in up and down scripts are ok

File: cmd/ocean/account.go (hunk 6 above)
Comment (sekulicd) [reply cmd/ocean/account.go:61]: Done

File: cmd/ocean/account.go (hunk 7 above)
Comment (sekulicd) [reply cmd/ocean/account.go]: Done

File: internal/core/application/account_service.go (hunk 8 above)
Comment (sekulicd) [reply internal/core/application/account_service.go:15]: Done

File: internal/core/application/account_service.go (hunk 9 above)
Comment (sekulicd) [reply internal/core/application/account_service.go]: Done

File: internal/core/domain/wallet.go (hunk 14 above)
Comment (sekulicd) [reply internal/core/domain/wallet.go]: Done

File: internal/core/domain/wallet.go (hunk 15 above)
Comment (sekulicd) [reply internal/core/domain/wallet.go]: Done

File: internal/core/domain/wallet_account.go (hunk 16 above)
Comment (sekulicd) [reply internal/core/domain/wallet_account.go]: Done

File: internal/core/domain/wallet_account.go (hunk 17 above)
Comment (sekulicd) [reply internal/core/domain/wallet_account.go]: Done

File: internal/core/domain/wallet_repository.go (hunk 18 above)
Comment (sekulicd) [reply internal/core/domain/wallet_repository.go]: Done

File: internal/infrastructure/storage/db/postgres/migration/20230112084523_accoun_table_update.down.sql (hunk 28 above)
Comment (sekulicd) [reply internal/infrastructure/storage/db/postgres/migration/20230112084523_accoun_table_update.down.sql:15]: Done

File: internal/infrastructure/storage/db/postgres/migration/20230112084523_accoun_table_update.up.sql [hunk 39]
```diff
@@ -0,0 +1,23 @@
...
+
+ALTER TABLE tx_input_account ADD COLUMN fk_account_namespace varchar(500);
+
+ALTER TABLE tx_input_account ADD FOREIGN KEY (fk_account_namespace) REFERENCES account(namespace);
+
+ALTER TABLE utxo DROP COLUMN account_name;
+
+ALTER TABLE utxo ADD COLUMN fk_account_namespace varchar(500);
+
+ALTER TABLE utxo ADD FOREIGN KEY (fk_account_namespace) REFERENCES account(namespace);
```
[the remark this answers is not in the document]
Comment (sekulicd) [reply internal/infrastructure/storage/db/postgres/migration/20230112084523_accoun_table_update.up.sql]: Done

File: internal/infrastructure/storage/db/postgres/sqlc/query.sql (hunk 29 above)
Comment (sekulicd) [reply internal/infrastructure/storage/db/postgres/sqlc/query.sql]: This is not possible without bigger change(using joins), I only updated GetAccount and used it to fetch account and use its namespace to get data from other tables.

Done

File: internal/infrastructure/storage/db/postgres/wallet_repository.go (hunk 30 above)
Comment (sekulicd) [reply internal/infrastructure/storage/db/postgres/wallet_repository.go]: This is ok behaviour as CreateAccount is also using UpdateWallet.
NACK

**Comment by sekulicd** [comment]
@altafan please check now.

**Comment by altafan** [comment]
This contains too many unnecessary changes. We can eventually recover the work done here to refactor the tests to open a dedicated PR. Closing in favour of #49.
